homeassistant/components/sensor/recorder.py

```diff
@@ -125,17 +125,17 @@
 WARN_UNSTABLE_UNIT = "sensor_warn_unstable_unit"
 
 
-def _get_entities(hass: HomeAssistant) -> list[tuple[str, str, str | None, str | None]]:
+def _get_entities(hass: HomeAssistant) -> list[tuple[str, State]]:
     """Get (entity_id, state_class, device_class) of all sensors for which to compile statistics."""
     all_sensors = hass.states.all(DOMAIN)
     entity_ids = []
 
     for state in all_sensors:
-        if (state_class := state.attributes.get(ATTR_STATE_CLASS)) not in STATE_CLASSES:
+        if not is_entity_recorded(hass, state.entity_id):
             continue
-        device_class = state.attributes.get(ATTR_DEVICE_CLASS)
-        unit = state.attributes.get(ATTR_UNIT_OF_MEASUREMENT)
-        entity_ids.append((state.entity_id, state_class, device_class, unit))
+        if (state.attributes.get(ATTR_STATE_CLASS)) not in STATE_CLASSES:
+            continue
+        entity_ids.append((state.entity_id, state))
```

Comment:
why even still store entity_id and not just state? Since entity ID is `state.entity_id`